src/middleware/validator.js

```diff
@@ -12,9 +12,19 @@ const performValidation = (ctx, schema) => {
     )
   }
 
-  if (!ctx || !ctx.request || !ctx.request.body) {
+  const dataToValidate = {}
+
+  if (ctx && ctx.request && ctx.request.body) {
+    dataToValidate.requestBody = ctx.request.body
+  }
+
+  if (ctx.lookupRequests) {
+    dataToValidate.lookupRequests = ctx.lookupRequests
+  }
+
+  if (!dataToValidate.requestBody && !dataToValidate.lookupRequests) {
     throw new Error(
-      `${ctx.state.metaData.name} (${ctx.state.uuid}): Invalid request body`
+      `${ctx.state.metaData.name} (${ctx.state.uuid}): No data to validate`
```

Comment:
Im wondering if this shouldnt rather be logging an error instead of throwing.

This logic seems correct in that, if the payload isnt supplied but validation needs to be executed then the request should fail as the validation wont be executed correctly.

However, i think there could be a case where a POST either includes are doesnt include data and the validation which is set as optional will only be applied if its supplied.

Maybe lets skip that use case for and update this at a later stage when we have a use case scenario adressing this

Reply:
Looking at the logic here it definitely seems like this should be a log not an error. Since the validation file was required it would have thrown an error before now if it were missing. And if the user supplies an empty object that's chilled too :| So I'm going to switch this to a warn log followed by a return so as to skip the rest of the validation logic

Reply:
Done in https://github.com/jembi/openhim-mediator-mapping/pull/37/commits/d78e810a7df6945856509627f11d271d99fb988d